internal/cmdupdate/cmdupdate.go

```diff
@@ -82,10 +88,10 @@ func (r *Runner) preRunE(_ *cobra.Command, args []string) error {
 		return err
 	}
 
-	r.Update.Path = string(p.DisplayPath)
+	relPath := string(p.DisplayPath)
```

Comment:
I considered adding this as part of the PR, but this would make an already massive PR even larger. This is mentioned in the PR description.
I do want to consolidate the code in `kpt/internal/pkg/pkg.go` with the update code, but it will take some effort. The new pkg code has added code that seems to duplicate existing functionality that needs to be cleaned up. Also, the package abstraction doesn't help when working with nested packages, which is where most of the filepath manipulation happens. In this case we do need to work with relative paths, but the `DisplayPath` abstraction doesn't really help.

I also have some concerns about how we currently handle relative and absolute paths in the `internal/pkg` code. We should translate any relative paths in the cobra commands and then only work with absolute paths. If user feedback makes more sense with relative paths, we should translate it back in the cobra commands. Kpt used to mix relative and absolute paths and it makes testing very difficult.

I guess this was just a long way of saying that I do plan on aligning with the package abstraction, but I want to do that separately. I also want to get this in first, so I can iterate on code that actually does the right thing instead of continue on working with the old design.

Reply:
Yes, computations should be based on `UniquePath`. `DisplayPath` is needed because different commands will display different information to users which is orthogonal to cobra or any choice of CLI framework for that matter. 

I'm not following why nested packages can't use package abstraction. Hydration logic already uses it in its recursive algorithm. But we can extend it with anything missing. It's a starting point for cleaning up the codebase, not exhaustive library.

I'm fine with deferring this in follow up with PRs to unblock things. I'm going to abstain from approving this and look at the follow up PRs instead.

Reply:
I'm not saying it is not possible, it just requires some work. Packaging is different from hydration in we are building and updating the package hierarchy as we are traversing it so it needs a slightly different approach.